modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,330 @@
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/body
+	greyscale_config_worn_digi = /datum/greyscale_config/shibari_worn_digi/body
+	greyscale_config_worn_taur_snake = /datum/greyscale_config/shibari_worn_taur_snake/body
+	greyscale_config_worn_taur_paw = /datum/greyscale_config/shibari_worn_taur_paw/body
+	greyscale_config_worn_taur_hoof = /datum/greyscale_config/shibari_worn_taur_hoof/body
+	greyscale_colors = "#bd8fcf"
+
+//processing stuff
+/obj/item/clothing/under/shibari/torso/process(delta_time)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_HIGH && hooman.pain < 30)
+		if(hooman.pain < 25)
+			hooman.adjustPain(0.6 * delta_time)
+		if(prob(10))
+			hooman.adjustOxyLoss(5)
+
+/obj/item/clothing/under/shibari/groin
+	name = "crotch rope shibari"
+	desc = "A rope that teases the wearer's genitals."
+	icon_state = "shibari_groin"
+
+	greyscale_config = /datum/greyscale_config/shibari_clothes/groin
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/groin
+	greyscale_config_worn_digi = /datum/greyscale_config/shibari_worn_digi/groin
+	greyscale_config_worn_taur_snake = /datum/greyscale_config/shibari_worn_taur_snake/groin
+	greyscale_config_worn_taur_paw = /datum/greyscale_config/shibari_worn_taur_paw/groin
+	greyscale_config_worn_taur_hoof = /datum/greyscale_config/shibari_worn_taur_hoof/groin
+	greyscale_colors = "#bd8fcf"
+
+//stuff to apply processing on equip and add mood event for perverts
+/obj/item/clothing/under/shibari/groin/equipped(mob/user, slot)
+	var/mob/living/carbon/human/hooman = user
+	if(hooman?.dna?.mutant_bodyparts["taur"])
+		slowdown = 4
+	else
+		slowdown = 0
+	. = ..()
+
+//processing stuff
+/obj/item/clothing/under/shibari/groin/process(delta_time)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_LOW && hooman.pleasure < 20)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_MED && hooman.pleasure < 60)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_HIGH)
+		hooman.adjustPleasure(0.6 * delta_time)
+
+/obj/item/clothing/under/shibari/full
+	name = "shibari fullbody ropes"
+	desc = "Bondage ropes that covers whole body"
+	icon_state = "shibari_fullbody"
+	rope_amount = 2
+
+	greyscale_config = /datum/greyscale_config/shibari_clothes/fullbody
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/fullbody
+	greyscale_config_worn_digi = /datum/greyscale_config/shibari_worn_digi/fullbody
+	greyscale_config_worn_taur_snake = /datum/greyscale_config/shibari_worn_taur_snake/fullbody
+	greyscale_config_worn_taur_paw = /datum/greyscale_config/shibari_worn_taur_paw/fullbody
+	greyscale_config_worn_taur_hoof = /datum/greyscale_config/shibari_worn_taur_hoof/fullbody
+	greyscale_colors = "#bd8fcf#bd8fcf"
+
+//processing stuff
+/obj/item/clothing/under/shibari/full/process(delta_time)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
```

Comment:
Why is this copypasted everywhere?

Reply:
again, each worn items behaves differently